nifi-nar-bundles/nifi-amqp-bundle/nifi-amqp-processors/src/main/java/org/apache/nifi/amqp/processors/PublishAMQP.java

```diff
@@ -108,10 +106,22 @@ public class PublishAMQP extends AbstractAMQPProcessor<AMQPPublisher> {
             .addValidator(StandardValidators.SINGLE_CHAR_VALIDATOR)
             .required(false)
             .build();
+
+    public static final PropertyDescriptor IGNORE_HEADER_WITH_NULL_VALUE = new PropertyDescriptor.Builder()
```

Comment:
Currently with 'Ignore Null Header = true', the processor throws an error, which is a little counter-intuitive because we told the processor to ignore the null header but it throws a warning?

Perhaps this would be better as 'Allow Null Header', and flipping the true/false condition, such that the warning message is thrown when 'Allow Null Header = false' and a null header is provided.

Reply:
The default value might also be 'true' to allow null headers by default? I'm actually not sure if there's a great reason not to allow null headers in the first place, so potentially this property isn't required.

Reply:
I agree with @thenatog that the property is not necessary and allowing the `null` header could be the only work mode. We did not introduce extra property for ignoring null headers in `ConsumeAMQP` either ([NIFI-10317](https://issues.apache.org/jira/browse/NIFI-10317)).